src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -37,13 +38,34 @@ void setUserStore(UserStore userStore) {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request,response);
+        //Checks whether user is logged in or not
+        String userName = (String) request.getSession().getAttribute("user");
+        System.out.println("The user store is " + userStore);
+        User loggedInUser = this.userStore.getUser(userName);
+        if (loggedInUser == null) {
+            response.sendRedirect("/login");
+            return;
+        }
+        System.out.println(loggedInUser.getDescription());
+        request.getSession().setAttribute("description", loggedInUser.getDescription());
+        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response)
             throws IOException, ServletException {
         String description = request.getParameter("description");
+        String userName = (String) request.getSession().getAttribute("user");
+        // read user from database using username
+        // get all the information of that loggedin user from datastore
+        User userToUpdate = this.userStore.getUser(userName);
+        //Setting Description of that Particular User in datastore
+        if (userToUpdate == null){
+            response.sendRedirect("/login");
+            return;
+        }
+        userToUpdate.setDescription(description);
```

Comment:
When getting user input we want to be careful on html pages for code injection (https://en.wikipedia.org/wiki/Code_injection). We should use the Jsoup helper class to sanitize the description before we save it anywhere. Check out ChatServlet#doPost (you dont need to do the whole "processString" part)

Reply:
Fixed it.